migen/build/lattice/icestorm.py

```diff
@@ -82,6 +82,23 @@ def _run_icestorm(build_name, source, yosys_opt, pnr_opt,
 
 
 class LatticeIceStormToolchain:
+    attr_translate = {
+        "keep": ("keep", "true"),
+        "no_retiming": None,  # yosys does not do retiming
+        "async_reg": None,
+
+        # The next 5 attributes are Vivado-specific. Ignore.
+        "mr_ff": None,
+        "mr_false_path": None,  # user-defined attribute
+        "ars_ff1": None,  # user-defined attribute
+        "ars_ff2": None,  # user-defined attribute
+        "ars_false_path": None,  # user-defined attribute
+
+        # Shift reg primitive issues are ISE-specific. Ignore.
```

Comment:
ditto

Reply:
Comment should probably be in this case "ice40 has no shift register primitives". Then "no_shreg_extract" can be safely ignored b/c "there's nothing to extract"?